wasatch/FeatureIdentificationDevice.py

```diff
@@ -767,10 +768,13 @@ def _read_eeprom(self) -> SpectrometerResponse:
                 log.error("exception reading upper_code 0x01 with page %d", page, exc_info=1)
             buf_len = 0 if buf is None else len(buf)
             if buf is None or len(buf) < 64:
-                msg = "unable to read EEPROM received buf of {buf} and len {len(buf)}"
+                msg = f"unable to read EEPROM received buf of {buf} and len {len(buf)}"
                 log.error(msg)
                 return SpectrometerResponse(False, error_lvl=ErrorLevel.medium,error_msg=msg)
             buffers.append(buf)
+        flat_buffers_all_ones = [byte == 0xff for page in buffers for byte in page]
```

Comment:
I've never actually seen a nested comprehension.  It doesn't look like I expected it to look.